src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/VerbParsers.tt

```diff
@@ -0,0 +1,107 @@
+<#@ template debug="false" hostspecific="true" language="C#" #>
+<#@ assembly name="System.Xml.XDocument" #>
+<#@ assembly name="System.Xml.Linq" #>
+<#@ assembly name="System.Xml" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Globalization" #>
+<#@ import namespace="System.Xml.Linq" #>
+<#@ import namespace="System.Linq" #>
+<#@ output extension=".cs" #>
+<#
+ string commandFile = this.Host.ResolvePath("Commands.xml");
+ XDocument commands = XDocument.Load(commandFile);
+#>
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+<#@ include file="Utils.tt" #>
+// Do not manually edit this autogenerated file:
+// instead modify the neighboring .tt file (text template) and/or NuGet.CommandLine.Xplat\Commands\SystemCommandLine\Commands.xml (data file),
+// then re-execute the text template via "run custom tool" on VS context menu for .tt file, or via dotnet-t4 global tool.
+
+using System;
+using System.CommandLine;
+using NuGet.Commands;
+using NuGet.Common;
+
+namespace NuGet.CommandLine.XPlat.Commands
+{
+<#
+    string getCommandHelpString(XElement cmd, string fallbackStringResource)
+    {
+        string help = cmd.Attribute(XName.Get("Help", ""))?.Value;
+
+        if (help != null)
+        {
+            return help;
+        }
+
+        return fallbackStringResource;
+    }
+
+// ********************** Verb-Parser Template
+var allElements = commands.Descendants(XName.Get("Verb","")).Concat(commands.Descendants(XName.Get("Command", "")));
+foreach (XElement verb in allElements)
+{
+    string commandName = verb.Attribute(XName.Get("Name", "")).Value;
+    string commandFormalName = InitCaps(commandName);
+    string commandFormalNameVar = commandFormalName + "Cmd";
+#>
+    internal partial class <#= commandFormalName #><#= verb.Name.LocalName #>Parser
+    {
+        internal static Func<ILogger> GetLoggerFunction;
+
+        internal static Command Register(Command app, Func<ILogger> getLogger)
+        {
+            var <#= commandFormalNameVar #> = new Command(name: "<#= commandName #>", description: Strings.<#= getCommandHelpString(verb, $"{commandFormalName}_Description") #>);
+
+            // Options directly under the verb '<#= commandName #>'
+<#
+    GenerateAddOptionOrArgument(verb, commandFormalNameVar, 12);
+
+    foreach (XElement noun in verb.Descendants(XName.Get("Noun")))
+    {
+        string nounName = noun.Attribute(XName.Get("Name", ""))?.Value;
+        string nounFormalName = InitCaps(nounName).Replace("-", "");
+        string nounFormalNameVar = nounFormalName + "Cmd";
+        if (nounFormalName == null)
+        {
+            throw new ArgumentException("nounFormalName cannot be null");
```

Comment:
This will never happen since if there are 2 cases above where it being `null` will throw.

Reply:
Actually, just one case. `+` concatenation is tolerant of `null`.

Reply:
Applied other suggestions. I think there's only one way to get a null by not having the xml Name attribute read at two lines above.